app/helpers/courses_helpers.rb

```diff
@@ -3,6 +3,25 @@ module Sinatra
   module UMDIO
     module Helpers
 
+      # generalize logic for checking if semester param valid
+      def check_semester app, coll_prefix
```

Comment:
`check_semester` should be a function of semester (i.e. take a semester argument)